modules/users/server/controllers/users.import.server.controller.js

```diff
@@ -0,0 +1,48 @@
+// const fs = require('fs');
+const path = require('path');
+const errorService = require(path.resolve('./modules/core/server/services/error.server.service'));
+const fileUpload = require(path.resolve('./modules/core/server/services/file-upload.service'));
+const decompress = require('decompress');
+const decompressUnzip = require('decompress-unzip');
+
+// log = require(path.resolve('./config/lib/logger')),
+
+/**
+ * Middleware to validate+process import file upload field
+ */
+exports.csImportUploadField = (req, res, next) => {
+  if (!req.user) {
+    return res.status(403).send({
+      message: errorService.getErrorMessageByKey('forbidden')
+    });
+  }
+
+  fileUpload.uploadFile(['application/zip'], 'import-file', req, res, next);
+};
+
+/**
+ *
+ */
+exports.processCsImport = async (req, res) => {
+
+  // eslint-disable-next-line
+  console.log(req.file.path);
+
+  try {
+    // @TODO: Need to have some good destination folder here
+    await decompress(req.file.path, './tmp/unzipped', {
```

Comment:
since the filename is random, maybe just `${filename}-output` as the directory name where files can be unzipped. Then after the process is done, delete both; temp file and the folder.